clar2wasm/src/wasm_utils.rs

```diff
@@ -1466,3 +1458,19 @@ fn reserve_space_for_return(
         }
     }
 }
+
+pub(crate) fn ordered_tuple_signature(
+    tuple: &TupleTypeSignature,
+) -> BTreeMap<&ClarityName, &TypeSignature> {
+    tuple.get_type_map().iter().collect()
+}
+
+pub(crate) fn owned_ordered_tuple_signature(
+    tuple: &TupleTypeSignature,
+) -> BTreeMap<ClarityName, TypeSignature> {
+    tuple
+        .get_type_map()
+        .iter()
+        .map(|(k, v)| (k.clone(), v.clone()))
```

Comment:
you can use `.cloned()` here instead of the explicit map

Reply:
No you cannot: `cloned` is for the cases where you yield references but here we have tuples of references.